Source/WebCore/loader/DocumentLoader.cpp

```diff
@@ -1000,14 +1001,17 @@ void DocumentLoader::responseReceived(ResourceResponse&& response, CompletionHan
     if (m_substituteData.isValid() || !platformStrategies()->loaderStrategy()->havePerformedSecurityChecks(response)) {
         auto url = response.url();
         RefPtr frame = m_frame.get();
-        ContentSecurityPolicy contentSecurityPolicy(URL { url }, this, frame->document());
+        ReportingClient* reportingClient = nullptr;
```

Comment:
This seems like a step in the wrong direction.  Why not just use a RefPtr<Document>?

Reply:
We don’t want to take ownership of the Document, so perhaps a WeakPtr. After looking at this yesterday, I think this argument to CSP was hacked in a bit poorly, and would need a larger patch to get this right.

I took the approach of matching the logic in the other overload of this method, and intend to follow up with a patch that addresses the ugly bare pointer used for this argument in current code.